dashboard/views.py

```diff
@@ -154,6 +154,45 @@ def get_queryset(self) -> QuerySet[Achievement]:
 		).order_by('-obtained', '-num_found')
 
 
+class AchievementLeaderboard(
+	LoginRequiredMixin, SuccessMessageMixin, UpdateView[Student, BaseModelForm[Student]]
+):
+	model = Student
+	fields = ('public', )
+	success_url = reverse_lazy('achievement-leaderboard')
+	object: Student
+	template_name = 'dashboard/achievement_leaderboard.html'
+
+	def form_valid(self, form: BaseModelForm[Student]):
+		messages.success(self.request, f"Updated settings for {self.object.user.username}!")
+		form.instance.user = self.request.user
```

Comment:
Add `assert isinstance(self.request.user, User)` above this to pass the type check.